harpy/scripts/demultiplex_gen1.py

```
@@ -0,0 +1,162 @@
+        "A" : read_barcodes(bx_a, "A"),
+        "B" : read_barcodes(bx_b, "B"),
+        "C" : read_barcodes(bx_c, "C"),
+        "D" : read_barcodes(bx_d, "D"),
+    }
+
+    #read schema
+    id_letter, samples_dict = read_schema(schema)
+    samples = list(set(samples_dict.values()))
+    samples.append("unidentified_data")
+    #create an array of files (one per sample) for writing
+    R1_output = {sample: open(f"{outdir}/{sample}.R1.fq", 'w') for sample in samples}
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)